// roy/src/parser.js
/* Jison generated parser */
var parser = (function(){
undefined
var parser = {trace: function trace() { },
yy: {},
symbols_: {"error":2,"program":3,"EOF":4,"body":5,"line":6,"TERMINATOR":7,"statement":8,"expression":9,"COMMENT":10,"block":11,"INDENT":12,"outdentOrEof":13,"doBody":14,"doLine":15,"IDENTIFIER":16,"LEFTARROW":17,"RETURN":18,"doBlock":19,"letFunction":20,"letBinding":21,"dataDecl":22,"typeDecl":23,"macro":24,"innerExpression":25,"LAMBDA":26,"paramList":27,"optType":28,"RIGHTARROW":29,"MATCH":30,"caseList":31,"DO":32,"call":33,"ifThenElse":34,"(":35,")":36,"&":37,"[|":38,"|]":39,"accessor":40,"!":41,"MATH":42,"CONCAT":43,"+":44,"-":45,"BOOLOP":46,"COMPARE":47,"WITH":48,"literal":49,"CASE":50,"pattern":51,"=":52,"innerPattern":53,"identifier":54,"patternIdentifiers":55,"IF":56,"THEN":57,"ELSE":58,"DATA":59,"optParamList":60,"dataList":61,"optTypeParamList":62,"|":63,"TYPE":64,"type":65,"{":66,"optTypePairs":67,"}":68,"typeParamList":69,"keywordOrIdentifier":70,":":71,",":72,"MACRO":73,"LET":74,"param":75,"argList":76,"tuple":77,"tupleList":78,"NUMBER":79,"STRING":80,"BOOLEAN":81,"[":82,"optValues":83,"]":84,"optPairs":85,"arrayValues":86,"keyPairs":87,"OUTDENT":88,"optTerm":89,".":90,"$accept":0,"$end":1},
terminals_: {2:"error",4:"EOF",7:"TERMINATOR",10:"COMMENT",12:"INDENT",16:"IDENTIFIER",17:"LEFTARROW",18:"RETURN",26:"LAMBDA",29:"RIGHTARROW",30:"MATCH",32:"DO",35:"(",36:")",37:"&",38:"[|",39:"|]",41:"!",42:"MATH",43:"CONCAT",44:"+",45:"-",46:"BOOLOP",47:"COMPARE",48:"WITH",50:"CASE",52:"=",56:"IF",57:"THEN",58:"ELSE",59:"DATA",63:"|",64:"TYPE",66:"{",68:"}",71:":",72:",",73:"MACRO",74:"LET",79:"NUMBER",80:"STRING",81:"BOOLEAN",82:"[",84:"]",88:"OUTDENT",90:"."},
productions_: [0,[3,1],[3,2],[5,1],[5,3],[5,2],[6,1],[6,1],[6,1],[11,3],[14,1],[14,3],[14,2],[15,1],[15,3],[15,2],[19,3],[8,1],[8,1],[8,1],[8,1],[8,1],[9,1],[9,5],[9,5],[9,5],[9,3],[9,1],[25,1],[25,3],[25,4],[25,3],[25,1],[25,3],[25,3],[25,3],[25,3],[25,3],[25,3],[25,3],[25,3],[25,1],[31,4],[31,6],[51,1],[51,1],[53,4],[55,1],[55,1],[55,2],[55,2],[34,7],[22,5],[22,7],[60,0],[60,1],[61,2],[61,4],[23,4],[65,2],[65,3],[62,0],[62,1],[69,1],[69,3],[69,2],[69,4],[67,0],[67,3],[67,5],[24,4],[24,4],[20,6],[20,6],[21,5],[21,7],[27,2],[27,1],[27,3],[27,2],[75,1],[75,5],[28,0],[28,2],[33,2],[33,4],[76,2],[76,1],[76,3],[76,2],[77,5],[78,1],[78,3],[49,1],[49,1],[49,1],[49,1],[49,3],[49,3],[83,0],[83,1],[86,1],[86,3],[85,0],[85,4],[85,1],[87,3],[87,5],[87,6],[89,0],[89,1],[40,1],[40,3],[40,5],[13,1],[13,1],[70,1],[70,1],[70,1],[70,1],[70,1],[70,1],[70,1],[70,1],[70,1],[70,1],[70,1],[54,1]],
performAction: function anonymous(yytext,yyleng,yylineno,yy,yystate,$$,_$) {

var $0 = $$.length - 1;
switch (yystate) {
case 1:return [];
break;
case 2:return $$[$0-1];
break;
case 3:this.$ = [$$[$0]];
break;
case 4:this.$ = $$[$0-2]; $$[$0-2].push($$[$0]);
break;
case 5:this.$ = $$[$0-1];
break;
case 6:this.$ = $$[$0];
break;
case 7:this.$ = $$[$0];
break;
case 8:this.$ = new yy.Comment($$[$0]);
break;
case 9:this.$ = $$[$0-1];
break;
case 10:this.$ = [$$[$0]];
break;
case 11:this.$ = $$[$0-2]; $$[$0-2].push($$[$0]);
break;
case 12:this.$ = $$[$0-1];
break;
case 13:this.$ = $$[$0];
break;
case 14:this.$ = new yy.Bind($$[$0-2], $$[$0]);
break;
case 15:this.$ = new yy.Return($$[$0]);
break;
case 16:this.$ = $$[$0-1];
break;
case 17:this.$ = $$[$0];
break;
case 18:this.$ = $$[$0];
break;
case 19:this.$ = $$[$0];
break;
case 20:this.$ = $$[$0];
break;
case 21:this.$ = $$[$0];
break;
case 22:this.$ = $$[$0];
break;
case 23:this.$ = new yy.Function(undefined, $$[$0-3], [$$[$0]], $$[$0-2]);
break;
case 24:this.$ = new yy.Function(undefined, $$[$0-3], $$[$0], $$[$0-2]);
break;
case 25:this.$ = new yy.Match($$[$0-3], $$[$0-1]);
break;
case 26:this.$ = new yy.Do($$[$0-1], $$[$0]);
break;
case 27:this.$ = $$[$0];
break;
case 28:this.$ = $$[$0];
break;
case 29:this.$ = new yy.Expression($$[$0-1]);
break;
case 30:this.$ = new yy.Replacement($$[$0-1]);
break;
case 31:this.$ = new yy.Quoted($$[$0-1]);
break;
case 32:this.$ = $$[$0];
break;
case 33:this.$ = new yy.Access($$[$0-2], $$[$0]);
break;
case 34:this.$ = new yy.BinaryNumberOperator($$[$0-1], $$[$0-2], $$[$0]);
break;
case 35:this.$ = new yy.BinaryStringOperator($$[$0-1], $$[$0-2], $$[$0]);
break;
case 36:this.$ = new yy.BinaryNumberOperator($$[$0-1], $$[$0-2], $$[$0]);
break;
case 37:this.$ = new yy.BinaryNumberOperator($$[$0-1], $$[$0-2], $$[$0]);
break;
case 38:this.$ = new yy.BinaryBooleanOperator($$[$0-1], $$[$0-2], $$[$0]);
break;
case 39:this.$ = new yy.BinaryGenericOperator($$[$0-1], $$[$0-2], $$[$0]);
break;
case 40:this.$ = new yy.With($$[$0-2], $$[$0]);
break;
case 41:this.$ = $$[$0];
break;
case 42:this.$ = [new yy.Case($$[$0-2], $$[$0])];
break;
case 43:this.$ = $$[$0-5]; $$[$0-5].push(new yy.Case($$[$0-2], $$[$0]));
break;
case 44:this.$ = $$[$0];
break;
case 45:this.$ = new yy.Pattern($$[$0], []);
break;
case 46:this.$ = new yy.Pattern($$[$0-2], $$[$0-1]);
break;
case 47:this.$ = [$$[$0]];
break;
case 48:this.$ = [$$[$0]];
break;
case 49:this.$ = $$[$0-1]; $$[$0-1].push($$[$0]);
break;
case 50:this.$ = $$[$0-1]; $$[$0-1].push($$[$0]);
break;
case 51:this.$ = new yy.IfThenElse($$[$0-5], $$[$0-3], $$[$0]);
break;
case 52:this.$ = new yy.Data($$[$0-3], $$[$0-2], $$[$0]);
break;
case 53:this.$ = new yy.Data($$[$0-5], $$[$0-4], $$[$0-1]);
break;
case 54:this.$ = [];
break;
case 55:this.$ = $$[$0];
break;
case 56:this.$ = [new yy.Tag($$[$0-1], $$[$0])];
break;
case 57:this.$ = $$[$0-3]; $$[$0-3].push(new yy.Tag($$[$0-1], $$[$0]));
break;
case 58:this.$ = new yy.Type($$[$0-2], $$[$0]);
break;
case 59:this.$ = new yy.TypeName($$[$0-1], $$[$0]);
break;
case 60:this.$ = new yy.TypeObject($$[$0-1]);
break;
case 61:this.$ = [];
break;
case 62:this.$ = $$[$0];
break;
case 63:this.$ = [new yy.TypeName($$[$0], [])];
break;
case 64:this.$ = [$$[$0-1]];
break;
case 65:this.$ = $$[$0-1]; $$[$0-1].push(new yy.TypeName($$[$0], []));
break;
case 66:this.$ = $$[$0-3]; $$[$0-3].push($$[$0-1]);
break;
case 67:this.$ = {};
break;
case 68:this.$ = {}; this.$[$$[$0-2]] = $$[$0];
break;
case 69:this.$ = $$[$0-4]; $$[$0-4][$$[$0-2]] = $$[$0];
break;
case 70:this.$ = new yy.Macro($$[$0-2], [$$[$0]]);
break;
case 71:this.$ = new yy.Macro($$[$0-2], $$[$0]);
break;
case 72:this.$ = new yy.Function($$[$0-4], $$[$0-3], $$[$0], $$[$0-2]);
break;
case 73:this.$ = new yy.Function($$[$0-4], $$[$0-3], [$$[$0]], $$[$0-2]);
break;
case 74:this.$ = new yy.Let($$[$0-3], $$[$0], $$[$0-2]);
break;
case 75:this.$ = new yy.Let($$[$0-5], $$[$0-1], $$[$0-4]);
break;
case 76:this.$ = [];
break;
case 77:this.$ = [$$[$0]];
break;
case 78:this.$ = $$[$0-2];
break;
case 79:this.$ = $$[$0-1]; $$[$0-1].push($$[$0]);
break;
case 80:this.$ = new yy.Arg($$[$0]);
break;
case 81:this.$ = new yy.Arg($$[$0-3], $$[$0-1]);
break;
case 82:
break;
case 83:this.$ = $$[$0]
break;
case 84:this.$ = new yy.Call($$[$0-1], $$[$0]);
break;
case 85:this.$ = new yy.Call($$[$0-2], $$[$0]);
break;
case 86:this.$ = [];
break;
case 87:this.$ = [$$[$0]];
break;
case 88:this.$ = $$[$0-2];
break;
case 89:this.$ = $$[$0-1]; $$[$0-1].push($$[$0]);
break;
case 90:$$[$0-1].unshift($$[$0-3]); this.$ = new yy.Tuple($$[$0-1]);
break;
case 91:this.$ = [$$[$0]];
break;
case 92:this.$ = $$[$0-2]; $$[$0-2].push($$[$0]);
break;
case 93:this.$ = new yy.Number($$[$0]);
break;
case 94:this.$ = new yy.String($$[$0]);
break;
case 95:this.$ = new yy.Boolean($$[$0]);
break;
case 96:this.$ = $$[$0];
break;
case 97:this.$ = new yy.Array($$[$0-1]);
break;
case 98:this.$ = new yy.Object($$[$0-1]);
break;
case 99:this.$ = [];
break;
case 100:this.$ = $$[$0];
break;
case 101:this.$ = [$$[$0]];
break;
case 102:this.$ = $$[$0-2]; $$[$0-2].push($$[$0]);
break;
case 103:this.$ = {};
break;
case 104:this.$ = $$[$0-2];
break;
case 105:this.$ = $$[$0];
break;
case 106:this.$ = {}; this.$[$$[$0-2]] = $$[$0];
break;
case 107:this.$ = $$[$0-4]; $$[$0-4][$$[$0-2]] = $$[$0];
break;
case 108:this.$ = $$[$0-5]; $$[$0-5][$$[$0-2]] = $$[$0];
break;
case 109:
break;
case 110:
break;
case 111:this.$ = new yy.Identifier($$[$0]);
break;
case 112:this.$ = new yy.Access($$[$0-2], $$[$0]);
break;
case 113:this.$ = new yy.Access($$[$0-3], $$[$0]);
break;
case 114:
break;
case 115:
break;
case 116:this.$ = $$[$0];
break;
case 117:this.$ = $$[$0];
break;
case 118:this.$ = $$[$0];
break;
case 119:this.$ = $$[$0];
break;
case 120:this.$ = $$[$0];
break;
case 121:this.$ = $$[$0];
break;
case 122:this.$ = $$[$0];
break;
case 123:this.$ = $$[$0];
break;
case 124:this.$ = $$[$0];
break;
case 125:this.$ = $$[$0];
break;
case 126:this.$ = $$[$0];
break;
case 127:this.$ = new yy.Identifier($$[$0]);
break;
}
},
table: [{3:1,4:[1,2],5:3,6:4,8:5,9:6,10:[1,7],16:[1,29],20:8,21:9,22:10,23:11,24:12,25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],59:[1,19],64:[1,20],66:[1,35],73:[1,21],74:[1,18],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{1:[3]},{1:[2,1]},{4:[1,36],7:[1,37]},{4:[2,3],7:[2,3],88:[2,3]},{4:[2,6],7:[2,6],88:[2,6]},{4:[2,7],7:[2,7],88:[2,7]},{4:[2,8],7:[2,8],88:[2,8]},{4:[2,17],7:[2,17],88:[2,17]},{4:[2,18],7:[2,18],88:[2,18]},{4:[2,19],7:[2,19],88:[2,19]},{4:[2,20],7:[2,20],88:[2,20]},{4:[2,21],7:[2,21],88:[2,21]},{4:[2,22],7:[2,22],36:[2,22],39:[2,22],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],68:[2,22],72:[2,22],84:[2,22],88:[2,22]},{16:[1,49],27:46,35:[1,47],75:48},{16:[1,29],25:50,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:53,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,27],7:[2,27],36:[2,27],39:[2,27],68:[2,27],72:[2,27],84:[2,27],88:[2,27]},{16:[1,54]},{16:[1,55]},{16:[1,56]},{16:[1,57]},{4:[2,28],7:[2,28],12:[2,28],16:[2,28],35:[2,28],36:[2,28],37:[2,28],38:[2,28],39:[2,28],41:[2,28],42:[2,28],43:[2,28],44:[2,28],45:[2,28],46:[2,28],47:[2,28],48:[2,28],56:[2,28],57:[2,28],66:[2,28],68:[2,28],72:[2,28],79:[2,28],80:[2,28],81:[2,28],82:[2,28],84:[2,28],88:[2,28]},{9:58,16:[1,29],25:59,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{35:[1,60]},{9:61,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,32],7:[2,32],16:[1,29],25:65,34:22,35:[1,64],36:[2,32],37:[1,24],38:[1,25],39:[2,32],40:52,41:[2,32],42:[2,32],43:[2,32],44:[2,32],45:[2,32],46:[2,32],47:[2,32],48:[2,32],49:27,56:[1,28],66:[1,35],68:[2,32],72:[2,32],76:62,77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],84:[2,32],88:[2,32],90:[1,63]},{4:[2,41],7:[2,41],12:[2,41],16:[2,41],35:[2,41],36:[2,41],37:[2,41],38:[2,41],39:[2,41],41:[2,41],42:[2,41],43:[2,41],44:[2,41],45:[2,41],46:[2,41],47:[2,41],48:[2,41],56:[2,41],57:[2,41],66:[2,41],68:[2,41],72:[2,41],79:[2,41],80:[2,41],81:[2,41],82:[2,41],84:[2,41],88:[2,41]},{16:[1,29],25:66,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,111],7:[2,111],12:[2,111],16:[2,111],35:[2,111],36:[2,111],37:[2,111],38:[2,111],39:[2,111],41:[2,111],42:[2,111],43:[2,111],44:[2,111],45:[2,111],46:[2,111],47:[2,111],48:[2,111],56:[2,111],57:[2,111],66:[2,111],68:[2,111],72:[2,111],79:[2,111],80:[2,111],81:[2,111],82:[2,111],84:[2,111],88:[2,111],90:[2,111]},{4:[2,93],7:[2,93],12:[2,93],16:[2,93],35:[2,93],36:[2,93],37:[2,93],38:[2,93],39:[2,93],41:[2,93],42:[2,93],43:[2,93],44:[2,93],45:[2,93],46:[2,93],47:[2,93],48:[2,93],56:[2,93],57:[2,93],66:[2,93],68:[2,93],72:[2,93],79:[2,93],80:[2,93],81:[2,93],82:[2,93],84:[2,93],88:[2,93]},{4:[2,94],7:[2,94],12:[2,94],16:[2,94],35:[2,94],36:[2,94],37:[2,94],38:[2,94],39:[2,94],41:[2,94],42:[2,94],43:[2,94],44:[2,94],45:[2,94],46:[2,94],47:[2,94],48:[2,94],56:[2,94],57:[2,94],66:[2,94],68:[2,94],72:[2,94],79:[2,94],80:[2,94],81:[2,94],82:[2,94],84:[2,94],88:[2,94]},{4:[2,95],7:[2,95],12:[2,95],16:[2,95],35:[2,95],36:[2,95],37:[2,95],38:[2,95],39:[2,95],41:[2,95],42:[2,95],43:[2,95],44:[2,95],45:[2,95],46:[2,95],47:[2,95],48:[2,95],56:[2,95],57:[2,95],66:[2,95],68:[2,95],72:[2,95],79:[2,95],80:[2,95],81:[2,95],82:[2,95],84:[2,95],88:[2,95]},{4:[2,96],7:[2,96],12:[2,96],16:[2,96],35:[2,96],36:[2,96],37:[2,96],38:[2,96],39:[2,96],41:[2,96],42:[2,96],43:[2,96],44:[2,96],45:[2,96],46:[2,96],47:[2,96],48:[2,96],56:[2,96],57:[2,96],66:[2,96],68:[2,96],72:[2,96],79:[2,96],80:[2,96],81:[2,96],82:[2,96],84:[2,96],88:[2,96]},{9:69,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],83:67,84:[2,99],86:68},{12:[1,71],16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],68:[2,103],70:73,73:[1,82],85:70,87:72},{1:[2,2]},{4:[2,5],6:85,7:[2,5],8:5,9:6,10:[1,7],16:[1,29],20:8,21:9,22:10,23:11,24:12,25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],59:[1,19],64:[1,20],66:[1,35],73:[1,21],74:[1,18],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],88:[2,5]},{16:[1,29],25:86,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:87,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:88,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:89,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:90,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:91,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:92,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,29],25:93,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,49],28:94,29:[2,82],35:[1,95],71:[1,97],75:96},{16:[1,99],36:[1,98]},{16:[2,77],29:[2,77],35:[2,77],52:[2,77],71:[2,77]},{16:[2,80],29:[2,80],35:[2,80],52:[2,80],71:[2,80]},{12:[1,100],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45]},{9:101,16:[1,29],25:59,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,32],7:[2,32],12:[2,32],16:[2,32],35:[2,32],36:[2,32],37:[2,32],38:[2,32],39:[2,32],41:[2,32],42:[2,32],43:[2,32],44:[2,32],45:[2,32],46:[2,32],47:[2,32],48:[2,32],56:[2,32],57:[2,32],66:[2,32],68:[2,32],72:[2,32],79:[2,32],80:[2,32],81:[2,32],82:[2,32],84:[2,32],88:[2,32],90:[1,63]},{12:[1,103],19:102,41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45]},{16:[1,49],27:104,28:105,35:[1,47],52:[2,82],71:[1,97],75:48},{16:[1,49],27:107,35:[1,47],52:[2,54],60:106,75:48},{52:[1,108]},{52:[1,109]},{36:[1,110]},{36:[2,22],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],72:[1,111]},{9:112,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{39:[1,113]},{4:[2,84],7:[2,84],16:[1,29],25:115,34:22,35:[1,114],36:[2,84],37:[1,24],38:[1,25],39:[2,84],40:52,49:27,56:[1,28],66:[1,35],68:[2,84],72:[2,84],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],84:[2,84],88:[2,84]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],70:116,73:[1,82]},{9:101,16:[1,29],25:59,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],36:[1,117],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,87],7:[2,87],16:[2,87],35:[2,87],36:[2,87],37:[2,87],38:[2,87],39:[2,87],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],56:[2,87],66:[2,87],68:[2,87],72:[2,87],79:[2,87],80:[2,87],81:[2,87],82:[2,87],84:[2,87],88:[2,87]},{41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],57:[1,118]},{84:[1,119]},{72:[1,120],84:[2,100]},{72:[2,101],84:[2,101]},{68:[1,121]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],70:73,73:[1,82],87:122},{7:[1,124],68:[2,105],72:[1,123]},{71:[1,125]},{4:[2,116],7:[2,116],12:[2,116],16:[2,116],35:[2,116],36:[2,116],37:[2,116],38:[2,116],39:[2,116],41:[2,116],42:[2,116],43:[2,116],44:[2,116],45:[2,116],46:[2,116],47:[2,116],48:[2,116],56:[2,116],57:[2,116],66:[2,116],68:[2,116],71:[2,116],72:[2,116],79:[2,116],80:[2,116],81:[2,116],82:[2,116],84:[2,116],88:[2,116],90:[2,116]},{4:[2,117],7:[2,117],12:[2,117],16:[2,117],35:[2,117],36:[2,117],37:[2,117],38:[2,117],39:[2,117],41:[2,117],42:[2,117],43:[2,117],44:[2,117],45:[2,117],46:[2,117],47:[2,117],48:[2,117],56:[2,117],57:[2,117],66:[2,117],68:[2,117],71:[2,117],72:[2,117],79:[2,117],80:[2,117],81:[2,117],82:[2,117],84:[2,117],88:[2,117],90:[2,117]},{4:[2,118],7:[2,118],12:[2,118],16:[2,118],35:[2,118],36:[2,118],37:[2,118],38:[2,118],39:[2,118],41:[2,118],42:[2,118],43:[2,118],44:[2,118],45:[2,118],46:[2,118],47:[2,118],48:[2,118],56:[2,118],57:[2,118],66:[2,118],68:[2,118],71:[2,118],72:[2,118],79:[2,118],80:[2,118],81:[2,118],82:[2,118],84:[2,118],88:[2,118],90:[2,118]},{4:[2,119],7:[2,119],12:[2,119],16:[2,119],35:[2,119],36:[2,119],37:[2,119],38:[2,119],39:[2,119],41:[2,119],42:[2,119],43:[2,119],44:[2,119],45:[2,119],46:[2,119],47:[2,119],48:[2,119],56:[2,119],57:[2,119],66:[2,119],68:[2,119],71:[2,119],72:[2,119],79:[2,119],80:[2,119],81:[2,119],82:[2,119],84:[2,119],88:[2,119],90:[2,119]},{4:[2,120],7:[2,120],12:[2,120],16:[2,120],35:[2,120],36:[2,120],37:[2,120],38:[2,120],39:[2,120],41:[2,120],42:[2,120],43:[2,120],44:[2,120],45:[2,120],46:[2,120],47:[2,120],48:[2,120],56:[2,120],57:[2,120],66:[2,120],68:[2,120],71:[2,120],72:[2,120],79:[2,120],80:[2,120],81:[2,120],82:[2,120],84:[2,120],88:[2,120],90:[2,120]},{4:[2,121],7:[2,121],12:[2,121],16:[2,121],35:[2,121],36:[2,121],37:[2,121],38:[2,121],39:[2,121],41:[2,121],42:[2,121],43:[2,121],44:[2,121],45:[2,121],46:[2,121],47:[2,121],48:[2,121],56:[2,121],57:[2,121],66:[2,121],68:[2,121],71:[2,121],72:[2,121],79:[2,121],80:[2,121],81:[2,121],82:[2,121],84:[2,121],88:[2,121],90:[2,121]},{4:[2,122],7:[2,122],12:[2,122],16:[2,122],35:[2,122],36:[2,122],37:[2,122],38:[2,122],39:[2,122],41:[2,122],42:[2,122],43:[2,122],44:[2,122],45:[2,122],46:[2,122],47:[2,122],48:[2,122],56:[2,122],57:[2,122],66:[2,122],68:[2,122],71:[2,122],72:[2,122],79:[2,122],80:[2,122],81:[2,122],82:[2,122],84:[2,122],88:[2,122],90:[2,122]},{4:[2,123],7:[2,123],12:[2,123],16:[2,123],35:[2,123],36:[2,123],37:[2,123],38:[2,123],39:[2,123],41:[2,123],42:[2,123],43:[2,123],44:[2,123],45:[2,123],46:[2,123],47:[2,123],48:[2,123],56:[2,123],57:[2,123],66:[2,123],68:[2,123],71:[2,123],72:[2,123],79:[2,123],80:[2,123],81:[2,123],82:[2,123],84:[2,123],88:[2,123],90:[2,123]},{4:[2,124],7:[2,124],12:[2,124],16:[2,124],35:[2,124],36:[2,124],37:[2,124],38:[2,124],39:[2,124],41:[2,124],42:[2,124],43:[2,124],44:[2,124],45:[2,124],46:[2,124],47:[2,124],48:[2,124],56:[2,124],57:[2,124],66:[2,124],68:[2,124],71:[2,124],72:[2,124],79:[2,124],80:[2,124],81:[2,124],82:[2,124],84:[2,124],88:[2,124],90:[2,124]},{4:[2,125],7:[2,125],12:[2,125],16:[2,125],35:[2,125],36:[2,125],37:[2,125],38:[2,125],39:[2,125],41:[2,125],42:[2,125],43:[2,125],44:[2,125],45:[2,125],46:[2,125],47:[2,125],48:[2,125],56:[2,125],57:[2,125],66:[2,125],68:[2,125],71:[2,125],72:[2,125],79:[2,125],80:[2,125],81:[2,125],82:[2,125],84:[2,125],88:[2,125],90:[2,125]},{4:[2,126],7:[2,126],12:[2,126],16:[2,126],35:[2,126],36:[2,126],37:[2,126],38:[2,126],39:[2,126],41:[2,126],42:[2,126],43:[2,126],44:[2,126],45:[2,126],46:[2,126],47:[2,126],48:[2,126],56:[2,126],57:[2,126],66:[2,126],68:[2,126],71:[2,126],72:[2,126],79:[2,126],80:[2,126],81:[2,126],82:[2,126],84:[2,126],88:[2,126],90:[2,126]},{4:[2,4],7:[2,4],88:[2,4]},{4:[2,33],7:[2,33],12:[2,33],16:[2,33],35:[2,33],36:[2,33],37:[2,33],38:[2,33],39:[2,33],41:[2,33],42:[1,39],43:[1,40],44:[2,33],45:[2,33],46:[2,33],47:[2,33],48:[2,33],56:[2,33],57:[2,33],66:[2,33],68:[2,33],72:[2,33],79:[2,33],80:[2,33],81:[2,33],82:[2,33],84:[2,33],88:[2,33]},{4:[2,34],7:[2,34],12:[2,34],16:[2,34],35:[2,34],36:[2,34],37:[2,34],38:[2,34],39:[2,34],41:[2,34],42:[2,34],43:[2,34],44:[2,34],45:[2,34],46:[2,34],47:[2,34],48:[2,34],56:[2,34],57:[2,34],66:[2,34],68:[2,34],72:[2,34],79:[2,34],80:[2,34],81:[2,34],82:[2,34],84:[2,34],88:[2,34]},{4:[2,35],7:[2,35],12:[2,35],16:[2,35],35:[2,35],36:[2,35],37:[2,35],38:[2,35],39:[2,35],41:[2,35],42:[2,35],43:[2,35],44:[2,35],45:[2,35],46:[2,35],47:[2,35],48:[2,35],56:[2,35],57:[2,35],66:[2,35],68:[2,35],72:[2,35],79:[2,35],80:[2,35],81:[2,35],82:[2,35],84:[2,35],88:[2,35]},{4:[2,36],7:[2,36],12:[2,36],16:[2,36],35:[2,36],36:[2,36],37:[2,36],38:[2,36],39:[2,36],41:[2,36],42:[1,39],43:[1,40],44:[2,36],45:[2,36],46:[2,36],47:[2,36],48:[2,36],56:[2,36],57:[2,36],66:[2,36],68:[2,36],72:[2,36],79:[2,36],80:[2,36],81:[2,36],82:[2,36],84:[2,36],88:[2,36]},{4:[2,37],7:[2,37],12:[2,37],16:[2,37],35:[2,37],36:[2,37],37:[2,37],38:[2,37],39:[2,37],41:[2,37],42:[1,39],43:[1,40],44:[2,37],45:[2,37],46:[2,37],47:[2,37],48:[2,37],56:[2,37],57:[2,37],66:[2,37],68:[2,37],72:[2,37],79:[2,37],80:[2,37],81:[2,37],82:[2,37],84:[2,37],88:[2,37]},{4:[2,38],7:[2,38],12:[2,38],16:[2,38],35:[2,38],36:[2,38],37:[2,38],38:[2,38],39:[2,38],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[2,38],47:[1,44],48:[1,45],56:[2,38],57:[2,38],66:[2,38],68:[2,38],72:[2,38],79:[2,38],80:[2,38],81:[2,38],82:[2,38],84:[2,38],88:[2,38]},{4:[2,39],7:[2,39],12:[2,39],16:[2,39],35:[2,39],36:[2,39],37:[2,39],38:[2,39],39:[2,39],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[2,39],47:[2,39],48:[2,39],56:[2,39],57:[2,39],66:[2,39],68:[2,39],72:[2,39],79:[2,39],80:[2,39],81:[2,39],82:[2,39],84:[2,39],88:[2,39]},{4:[2,40],7:[2,40],12:[2,40],16:[2,40],35:[2,40],36:[2,40],37:[2,40],38:[2,40],39:[2,40],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[2,40],47:[2,40],48:[2,40],56:[2,40],57:[2,40],66:[2,40],68:[2,40],72:[2,40],79:[2,40],80:[2,40],81:[2,40],82:[2,40],84:[2,40],88:[2,40]},{29:[1,126]},{16:[1,99],36:[1,127]},{16:[2,79],29:[2,79],35:[2,79],52:[2,79],71:[2,79]},{16:[1,129],65:128,66:[1,130]},{16:[2,76],29:[2,76],35:[2,76],52:[2,76],71:[2,76]},{71:[1,131]},{31:132,50:[1,133]},{36:[1,134]},{4:[2,26],7:[2,26],36:[2,26],39:[2,26],68:[2,26],72:[2,26],84:[2,26],88:[2,26]},{6:137,8:5,9:6,10:[1,7],14:135,15:136,16:[1,138],18:[1,139],20:8,21:9,22:10,23:11,24:12,25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],59:[1,19],64:[1,20],66:[1,35],73:[1,21],74:[1,18],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,49],28:140,35:[1,95],52:[2,82],71:[1,97],75:96},{52:[1,141]},{52:[1,142]},{16:[1,49],35:[1,95],52:[2,55],75:96},{16:[1,129],65:143,66:[1,130]},{9:144,11:145,12:[1,146],16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,29],7:[2,29],16:[1,29],25:65,34:22,35:[1,64],36:[2,29],37:[1,24],38:[1,25],39:[2,29],40:52,41:[2,29],42:[2,29],43:[2,29],44:[2,29],45:[2,29],46:[2,29],47:[2,29],48:[2,29],49:27,56:[1,28],66:[1,35],68:[2,29],72:[2,29],76:147,77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],84:[2,29],88:[2,29],90:[1,148]},{16:[1,29],25:150,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,78:149,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{36:[1,151]},{4:[2,31],7:[2,31],12:[2,31],16:[2,31],35:[2,31],36:[2,31],37:[2,31],38:[2,31],39:[2,31],41:[2,31],42:[2,31],43:[2,31],44:[2,31],45:[2,31],46:[2,31],47:[2,31],48:[2,31],56:[2,31],57:[2,31],66:[2,31],68:[2,31],72:[2,31],79:[2,31],80:[2,31],81:[2,31],82:[2,31],84:[2,31],88:[2,31]},{9:101,16:[1,29],25:59,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],36:[1,152],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,89],7:[2,89],16:[2,89],35:[2,89],36:[2,89],37:[2,89],38:[2,89],39:[2,89],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],56:[2,89],66:[2,89],68:[2,89],72:[2,89],79:[2,89],80:[2,89],81:[2,89],82:[2,89],84:[2,89],88:[2,89]},{4:[2,112],7:[2,112],12:[2,112],16:[2,112],35:[2,112],36:[2,112],37:[2,112],38:[2,112],39:[2,112],41:[2,112],42:[2,112],43:[2,112],44:[2,112],45:[2,112],46:[2,112],47:[2,112],48:[2,112],56:[2,112],57:[2,112],66:[2,112],68:[2,112],72:[2,112],79:[2,112],80:[2,112],81:[2,112],82:[2,112],84:[2,112],88:[2,112],90:[2,112]},{4:[2,86],7:[2,86],16:[2,86],35:[2,86],36:[2,86],37:[2,86],38:[2,86],39:[2,86],56:[2,86],66:[2,86],68:[2,86],72:[2,86],79:[2,86],80:[2,86],81:[2,86],82:[2,86],84:[2,86],88:[2,86]},{11:153,12:[1,146]},{4:[2,97],7:[2,97],12:[2,97],16:[2,97],35:[2,97],36:[2,97],37:[2,97],38:[2,97],39:[2,97],41:[2,97],42:[2,97],43:[2,97],44:[2,97],45:[2,97],46:[2,97],47:[2,97],48:[2,97],56:[2,97],57:[2,97],66:[2,97],68:[2,97],72:[2,97],79:[2,97],80:[2,97],81:[2,97],82:[2,97],84:[2,97],88:[2,97]},{9:154,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,98],7:[2,98],12:[2,98],16:[2,98],35:[2,98],36:[2,98],37:[2,98],38:[2,98],39:[2,98],41:[2,98],42:[2,98],43:[2,98],44:[2,98],45:[2,98],46:[2,98],47:[2,98],48:[2,98],56:[2,98],57:[2,98],66:[2,98],68:[2,98],72:[2,98],79:[2,98],80:[2,98],81:[2,98],82:[2,98],84:[2,98],88:[2,98]},{7:[1,124],72:[1,123],88:[1,155]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],70:156,73:[1,82]},{7:[1,158],16:[2,109],18:[2,109],30:[2,109],32:[2,109],48:[2,109],50:[2,109],57:[2,109],58:[2,109],59:[2,109],64:[2,109],73:[2,109],89:157},{9:159,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{9:160,11:161,12:[1,146],16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[2,78],29:[2,78],35:[2,78],52:[2,78],71:[2,78]},{29:[2,83],52:[2,83]},{4:[2,61],7:[2,61],16:[1,164],29:[2,61],35:[1,165],36:[2,61],52:[2,61],62:162,68:[2,61],69:163,72:[2,61],88:[2,61]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],67:166,68:[2,67],70:167,72:[2,67],73:[1,82]},{16:[1,129],65:168,66:[1,130]},{4:[1,172],7:[1,170],13:169,88:[1,171]},{16:[1,177],35:[1,176],51:173,53:174,54:175},{4:[2,29],7:[2,29],12:[2,29],16:[2,29],35:[2,29],36:[2,29],37:[2,29],38:[2,29],39:[2,29],41:[2,29],42:[2,29],43:[2,29],44:[2,29],45:[2,29],46:[2,29],47:[2,29],48:[2,29],56:[2,29],57:[2,29],66:[2,29],68:[2,29],72:[2,29],79:[2,29],80:[2,29],81:[2,29],82:[2,29],84:[2,29],88:[2,29],90:[1,148]},{4:[1,172],7:[1,179],13:178,88:[1,171]},{4:[2,10],7:[2,10],88:[2,10]},{4:[2,13],7:[2,13],88:[2,13]},{4:[2,111],7:[2,111],16:[2,111],17:[1,180],35:[2,111],37:[2,111],38:[2,111],41:[2,111],42:[2,111],43:[2,111],44:[2,111],45:[2,111],46:[2,111],47:[2,111],48:[2,111],56:[2,111],66:[2,111],79:[2,111],80:[2,111],81:[2,111],82:[2,111],88:[2,111],90:[2,111]},{9:181,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{52:[1,182]},{9:183,12:[1,184],16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{12:[1,186],16:[1,187],61:185},{4:[2,58],7:[2,58],88:[2,58]},{4:[2,70],7:[2,70],88:[2,70]},{4:[2,71],7:[2,71],88:[2,71]},{5:188,6:4,8:5,9:6,10:[1,7],16:[1,29],20:8,21:9,22:10,23:11,24:12,25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],59:[1,19],64:[1,20],66:[1,35],73:[1,21],74:[1,18],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,85],7:[2,85],16:[1,29],25:115,34:22,35:[1,114],36:[2,85],37:[1,24],38:[1,25],39:[2,85],40:52,49:27,56:[1,28],66:[1,35],68:[2,85],72:[2,85],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],84:[2,85],88:[2,85]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],70:189,73:[1,82]},{36:[1,190],72:[1,191]},{36:[2,91],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],72:[2,91]},{4:[2,30],7:[2,30],12:[2,30],16:[2,30],35:[2,30],36:[2,30],37:[2,30],38:[2,30],39:[2,30],41:[2,30],42:[2,30],43:[2,30],44:[2,30],45:[2,30],46:[2,30],47:[2,30],48:[2,30],56:[2,30],57:[2,30],66:[2,30],68:[2,30],72:[2,30],79:[2,30],80:[2,30],81:[2,30],82:[2,30],84:[2,30],88:[2,30]},{4:[2,88],7:[2,88],16:[2,88],35:[2,88],36:[2,88],37:[2,88],38:[2,88],39:[2,88],56:[2,88],66:[2,88],68:[2,88],72:[2,88],79:[2,88],80:[2,88],81:[2,88],82:[2,88],84:[2,88],88:[2,88]},{7:[1,192]},{72:[2,102],84:[2,102]},{7:[1,193]},{71:[1,194]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],70:195,73:[1,82]},{16:[2,110],18:[2,110],30:[2,110],32:[2,110],48:[2,110],50:[2,110],57:[2,110],58:[2,110],59:[2,110],64:[2,110],73:[2,110]},{7:[2,106],68:[2,106],72:[2,106],88:[2,106]},{4:[2,23],7:[2,23],36:[2,23],39:[2,23],68:[2,23],72:[2,23],84:[2,23],88:[2,23]},{4:[2,24],7:[2,24],36:[2,24],39:[2,24],68:[2,24],72:[2,24],84:[2,24],88:[2,24]},{4:[2,59],7:[2,59],29:[2,59],36:[2,59],52:[2,59],68:[2,59],72:[2,59],88:[2,59]},{4:[2,62],7:[2,62],16:[1,196],29:[2,62],35:[1,197],36:[2,62],52:[2,62],63:[2,62],68:[2,62],72:[2,62],88:[2,62]},{4:[2,63],7:[2,63],16:[2,63],29:[2,63],35:[2,63],36:[2,63],52:[2,63],63:[2,63],68:[2,63],72:[2,63],88:[2,63]},{16:[1,129],65:198,66:[1,130]},{68:[1,199],72:[1,200]},{71:[1,201]},{36:[1,202]},{4:[2,25],7:[2,25],36:[2,25],39:[2,25],68:[2,25],72:[2,25],84:[2,25],88:[2,25]},{50:[1,203]},{4:[2,114],7:[2,114],12:[2,114],16:[2,114],35:[2,114],36:[2,114],37:[2,114],38:[2,114],39:[2,114],41:[2,114],42:[2,114],43:[2,114],44:[2,114],45:[2,114],46:[2,114],47:[2,114],48:[2,114],56:[2,114],57:[2,114],66:[2,114],68:[2,114],72:[2,114],79:[2,114],80:[2,114],81:[2,114],82:[2,114],84:[2,114],88:[2,114]},{4:[2,115],7:[2,115],12:[2,115],16:[2,115],35:[2,115],36:[2,115],37:[2,115],38:[2,115],39:[2,115],41:[2,115],42:[2,115],43:[2,115],44:[2,115],45:[2,115],46:[2,115],47:[2,115],48:[2,115],56:[2,115],57:[2,115],66:[2,115],68:[2,115],72:[2,115],79:[2,115],80:[2,115],81:[2,115],82:[2,115],84:[2,115],88:[2,115]},{52:[1,204]},{52:[2,44]},{52:[2,45]},{16:[1,177],54:205},{16:[2,127],35:[2,127],36:[2,127],52:[2,127]},{4:[2,16],7:[2,16],36:[2,16],39:[2,16],68:[2,16],72:[2,16],84:[2,16],88:[2,16]},{4:[2,12],6:137,7:[2,12],8:5,9:6,10:[1,7],15:206,16:[1,138],18:[1,139],20:8,21:9,22:10,23:11,24:12,25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],59:[1,19],64:[1,20],66:[1,35],73:[1,21],74:[1,18],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34],88:[2,12]},{9:207,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,15],7:[2,15],88:[2,15]},{9:209,11:208,12:[1,146],16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,74],7:[2,74],88:[2,74]},{9:210,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{4:[2,52],7:[2,52],63:[1,211],88:[2,52]},{16:[1,187],61:212},{4:[2,61],7:[2,61],16:[1,164],35:[1,165],62:213,63:[2,61],69:163,88:[2,61]},{4:[1,172],7:[1,37],13:214,88:[1,171]},{4:[2,113],7:[2,113],12:[2,113],16:[2,113],35:[2,113],36:[2,113],37:[2,113],38:[2,113],39:[2,113],41:[2,113],42:[2,113],43:[2,113],44:[2,113],45:[2,113],46:[2,113],47:[2,113],48:[2,113],56:[2,113],57:[2,113],66:[2,113],68:[2,113],72:[2,113],79:[2,113],80:[2,113],81:[2,113],82:[2,113],84:[2,113],88:[2,113],90:[2,113]},{4:[2,90],7:[2,90],12:[2,90],16:[2,90],35:[2,90],36:[2,90],37:[2,90],38:[2,90],39:[2,90],41:[2,90],42:[2,90],43:[2,90],44:[2,90],45:[2,90],46:[2,90],47:[2,90],48:[2,90],56:[2,90],57:[2,90],66:[2,90],68:[2,90],72:[2,90],79:[2,90],80:[2,90],81:[2,90],82:[2,90],84:[2,90],88:[2,90]},{16:[1,29],25:215,34:22,35:[1,51],37:[1,24],38:[1,25],40:52,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{58:[1,216]},{68:[2,104]},{9:217,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{71:[1,218]},{4:[2,65],7:[2,65],16:[2,65],29:[2,65],35:[2,65],36:[2,65],52:[2,65],63:[2,65],68:[2,65],72:[2,65],88:[2,65]},{16:[1,129],65:219,66:[1,130]},{36:[1,220]},{4:[2,60],7:[2,60],29:[2,60],36:[2,60],52:[2,60],68:[2,60],72:[2,60],88:[2,60]},{16:[1,84],18:[1,81],30:[1,78],32:[1,80],48:[1,83],50:[1,79],57:[1,74],58:[1,75],59:[1,76],64:[1,77],70:221,73:[1,82]},{16:[1,129],65:222,66:[1,130]},{16:[2,81],29:[2,81],35:[2,81],52:[2,81],71:[2,81]},{16:[1,177],35:[1,176],51:223,53:174,54:175},{9:224,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[1,177],35:[1,176],53:227,54:226,55:225},{4:[2,11],7:[2,11],88:[2,11]},{4:[2,14],7:[2,14],88:[2,14]},{4:[2,72],7:[2,72],88:[2,72]},{4:[2,73],7:[2,73],88:[2,73]},{4:[1,172],13:228,88:[1,171]},{16:[1,229]},{4:[1,172],13:230,63:[1,211],88:[1,171]},{4:[2,56],7:[2,56],63:[2,56],88:[2,56]},{4:[2,9],7:[2,9],12:[2,9],16:[2,9],35:[2,9],36:[2,9],37:[2,9],38:[2,9],39:[2,9],41:[2,9],42:[2,9],43:[2,9],44:[2,9],45:[2,9],46:[2,9],47:[2,9],48:[2,9],56:[2,9],57:[2,9],66:[2,9],68:[2,9],72:[2,9],79:[2,9],80:[2,9],81:[2,9],82:[2,9],84:[2,9],88:[2,9]},{36:[2,92],41:[1,38],42:[1,39],43:[1,40],44:[1,41],45:[1,42],46:[1,43],47:[1,44],48:[1,45],72:[2,92]},{11:231,12:[1,146]},{7:[2,107],68:[2,107],72:[2,107],88:[2,107]},{9:232,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{36:[1,233]},{4:[2,64],7:[2,64],16:[2,64],29:[2,64],35:[2,64],36:[2,64],52:[2,64],63:[2,64],68:[2,64],72:[2,64],88:[2,64]},{71:[1,234]},{68:[2,68],72:[2,68]},{52:[1,235]},{4:[2,42],7:[2,42],88:[2,42]},{16:[1,177],35:[1,176],36:[1,236],53:237,54:238},{16:[2,47],35:[2,47],36:[2,47]},{16:[2,48],35:[2,48],36:[2,48]},{4:[2,75],7:[2,75],88:[2,75]},{4:[2,61],7:[2,61],16:[1,164],35:[1,165],62:239,63:[2,61],69:163,88:[2,61]},{4:[2,53],7:[2,53],88:[2,53]},{4:[2,51],7:[2,51],12:[2,51],16:[2,51],35:[2,51],36:[2,51],37:[2,51],38:[2,51],39:[2,51],41:[2,51],42:[2,51],43:[2,51],44:[2,51],45:[2,51],46:[2,51],47:[2,51],48:[2,51],56:[2,51],57:[2,51],66:[2,51],68:[2,51],72:[2,51],79:[2,51],80:[2,51],81:[2,51],82:[2,51],84:[2,51],88:[2,51]},{7:[2,108],68:[2,108],72:[2,108],88:[2,108]},{4:[2,66],7:[2,66],16:[2,66],29:[2,66],35:[2,66],36:[2,66],52:[2,66],63:[2,66],68:[2,66],72:[2,66],88:[2,66]},{16:[1,129],65:240,66:[1,130]},{9:241,16:[1,29],25:13,26:[1,14],30:[1,15],32:[1,16],33:17,34:22,35:[1,23],37:[1,24],38:[1,25],40:26,49:27,56:[1,28],66:[1,35],77:33,79:[1,30],80:[1,31],81:[1,32],82:[1,34]},{16:[2,46],35:[2,46],36:[2,46],52:[2,46]},{16:[2,49],35:[2,49],36:[2,49]},{16:[2,50],35:[2,50],36:[2,50]},{4:[2,57],7:[2,57],63:[2,57],88:[2,57]},{68:[2,69],72:[2,69]},{4:[2,43],7:[2,43],88:[2,43]}],
defaultActions: {2:[2,1],36:[2,2],174:[2,44],175:[2,45],193:[2,104]},
parseError: function parseError(str, hash) {
    throw new Error(str);
},
parse: function parse(input) {
    var self = this,
        stack = [0],
        vstack = [null], // semantic value stack
        lstack = [], // location stack
        table = this.table,
        yytext = '',
        yylineno = 0,
        yyleng = 0,
        recovering = 0,
        TERROR = 2,
        EOF = 1;

    //this.reductionCount = this.shiftCount = 0;

    this.lexer.setInput(input);
    this.lexer.yy = this.yy;
    this.yy.lexer = this.lexer;
    if (typeof this.lexer.yylloc == 'undefined')
        this.lexer.yylloc = {};
    var yyloc = this.lexer.yylloc;
    lstack.push(yyloc);

    if (typeof this.yy.parseError === 'function')
        this.parseError = this.yy.parseError;

    function popStack (n) {
        stack.length = stack.length - 2*n;
        vstack.length = vstack.length - n;
        lstack.length = lstack.length - n;
    }

    function lex() {
        var token;
        token = self.lexer.lex() || 1; // $end = 1
        // if token isn't its numeric value, convert
        if (typeof token !== 'number') {
            token = self.symbols_[token] || token;
        }
        return token;
    }

    var symbol, preErrorSymbol, state, action, a, r, yyval={},p,len,newState, expected;
    while (true) {
        // retreive state number from top of stack
        state = stack[stack.length-1];

        // use default actions if available
        if (this.defaultActions[state]) {
            action = this.defaultActions[state];
        } else {
            if (symbol == null)
                symbol = lex();
            // read action for current state and first input
            action = table[state] && table[state][symbol];
        }

        // handle parse error
        _handle_error:
        if (typeof action === 'undefined' || !action.length || !action[0]) {

            if (!recovering) {
                // Report error
                expected = [];
                for (p in table[state]) if (this.terminals_[p] && p > 2) {
                    expected.push("'"+this.terminals_[p]+"'");
                }
                var errStr = '';
                if (this.lexer.showPosition) {
                    errStr = 'Parse error on line '+(yylineno+1)+":\n"+this.lexer.showPosition()+"\nExpecting "+expected.join(', ') + ", got '" + this.terminals_[symbol]+ "'";
                } else {
                    errStr = 'Parse error on line '+(yylineno+1)+": Unexpected " +
                                  (symbol == 1 /*EOF*/ ? "end of input" :
                                              ("'"+(this.terminals_[symbol] || symbol)+"'"));
                }
                this.parseError(errStr,
                    {text: this.lexer.match, token: this.terminals_[symbol] || symbol, line: this.lexer.yylineno, loc: yyloc, expected: expected});
            }

            // just recovered from another error
            if (recovering == 3) {
                if (symbol == EOF) {
                    throw new Error(errStr || 'Parsing halted.');
                }

                // discard current lookahead and grab another
                yyleng = this.lexer.yyleng;
                yytext = this.lexer.yytext;
                yylineno = this.lexer.yylineno;
                yyloc = this.lexer.yylloc;
                symbol = lex();
            }

            // try to recover from error
            while (1) {
                // check for error recovery rule in this state
                if ((TERROR.toString()) in table[state]) {
                    break;
                }
                if (state == 0) {
                    throw new Error(errStr || 'Parsing halted.');
                }
                popStack(1);
                state = stack[stack.length-1];
            }

            preErrorSymbol = symbol; // save the lookahead token
            symbol = TERROR;         // insert generic error symbol as new lookahead
            state = stack[stack.length-1];
            action = table[state] && table[state][TERROR];
            recovering = 3; // allow 3 real symbols to be shifted before reporting a new error
        }

        // this shouldn't happen, unless resolve defaults are off
        if (action[0] instanceof Array && action.length > 1) {
            throw new Error('Parse Error: multiple actions possible at state: '+state+', token: '+symbol);
        }

        switch (action[0]) {

            case 1: // shift
                //this.shiftCount++;

                stack.push(symbol);
                vstack.push(this.lexer.yytext);
                lstack.push(this.lexer.yylloc);
                stack.push(action[1]); // push state
                symbol = null;
                if (!preErrorSymbol) { // normal execution/no error
                    yyleng = this.lexer.yyleng;
                    yytext = this.lexer.yytext;
                    yylineno = this.lexer.yylineno;
                    yyloc = this.lexer.yylloc;
                    if (recovering > 0)
                        recovering--;
                } else { // error just occurred, resume old lookahead f/ before error
                    symbol = preErrorSymbol;
                    preErrorSymbol = null;
                }
                break;

            case 2: // reduce
                //this.reductionCount++;

                len = this.productions_[action[1]][1];

                // perform semantic action
                yyval.$ = vstack[vstack.length-len]; // default to $$ = $1
                // default location, uses first token for firsts, last for lasts
                yyval._$ = {
                    first_line: lstack[lstack.length-(len||1)].first_line,
                    last_line: lstack[lstack.length-1].last_line,
                    first_column: lstack[lstack.length-(len||1)].first_column,
                    last_column: lstack[lstack.length-1].last_column
                };
                r = this.performAction.call(yyval, yytext, yyleng, yylineno, this.yy, action[1], vstack, lstack);

                if (typeof r !== 'undefined') {
                    return r;
                }

                // pop off stack
                if (len) {
                    stack = stack.slice(0,-1*len*2);
                    vstack = vstack.slice(0, -1*len);
                    lstack = lstack.slice(0, -1*len);
                }

                stack.push(this.productions_[action[1]][0]);    // push nonterminal (reduce)
                vstack.push(yyval.$);
                lstack.push(yyval._$);
                // goto new state = table[STATE][NONTERMINAL]
                newState = table[stack[stack.length-2]][stack[stack.length-1]];
                stack.push(newState);
                break;

            case 3: // accept
                return true;
        }

    }

    return true;
}};
return parser;
})();
if (typeof require !== 'undefined' && typeof exports !== 'undefined') {
exports.parser = parser;
exports.parse = function () { return parser.parse.apply(parser, arguments); }
exports.main = function commonjsMain(args) {
    if (!args[1])
        throw new Error('Usage: '+args[0]+' FILE');
    if (typeof process !== 'undefined') {
        var source = require('fs').readFileSync(require('path').join(process.cwd(), args[1]), "utf8");
    } else {
        var cwd = require("file").path(require("file").cwd());
        var source = cwd.join(args[1]).read({charset: "utf-8"});
    }
    return exports.parser.parse(source);
}
if (typeof module !== 'undefined' && require.main === module) {
  exports.main(typeof process !== 'undefined' ? process.argv.slice(1) : require("system").args);
}
}
